networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/encodings/compression/snappy/SnappyFrameEncoder.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.eth2.rpc.core.encodings.compression.snappy;
+
+import static tech.pegasys.teku.networking.eth2.rpc.core.encodings.compression.snappy.SnappyUtil.calculateChecksum;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.compression.Snappy;
+import org.apache.tuweni.bytes.Bytes;
+import tech.pegasys.teku.networking.eth2.rpc.core.encodings.compression.exceptions.CompressionException;
+
+/**
+ * This class is mostly borrowed from the Netty implementation:
+ * https://github.com/netty/netty/blob/4.1/codec/src/main/java/io/netty/handler/codec/compression/SnappyFramedEncoder.java
+ *
+ * <p>Compresses a {@link ByteBuf} using the Snappy framing format.
+ *
+ * <p>See <a href="https://github.com/google/snappy/blob/master/framing_format.txt">Snappy framing
+ * format</a>.
+ */
+public class SnappyFrameEncoder {
+  /**
+   * The minimum amount that we'll consider actually attempting to compress. This value is preamble
+   * + the minimum length our Snappy service will compress (instead of just emitting a literal).
+   */
+  private static final int MIN_COMPRESSIBLE_LENGTH = 18;
+
+  /**
+   * All streams should start with the "Stream identifier", containing chunk type 0xff, a length
+   * field of 0x6, and 'sNaPpY' in ASCII.
+   */
+  private static final byte[] STREAM_START = {
+    (byte) 0xff, 0x06, 0x00, 0x00, 0x73, 0x4e, 0x61, 0x50, 0x70, 0x59
+  };
+
+  private final Snappy snappy = new Snappy();
+  private boolean started;
+
+  public Bytes encode(Bytes in) {
+    ByteBuf inBuf = Unpooled.wrappedBuffer(in.toArrayUnsafe());
+    ByteBuf outBuf = Unpooled.buffer(in.size() / 2);
+    try {
+      encode(inBuf, outBuf);
+      byte[] bytes = new byte[outBuf.readableBytes()];
+      outBuf.readBytes(bytes);
+      return Bytes.wrap(bytes);
+    } finally {
+      inBuf.release();
+      outBuf.release();
+    }
+  }
+
+  public void encode(ByteBuf in, ByteBuf out) {
+    if (!in.isReadable()) {
+      return;
+    }
+
+    if (!started) {
+      started = true;
+      out.writeBytes(STREAM_START);
+    }
+
+    int dataLength = in.readableBytes();
+    if (dataLength > MIN_COMPRESSIBLE_LENGTH) {
+      for (; ; ) {
```

Comment:
Never seen this 🤔 What about just:
```suggestion
      while(true) {
```

Reply:
Fixed. 
This is from original code and I suppose it's a C style